src/core/models/appliedhighlightedfields.js

```diff
@@ -0,0 +1,56 @@
+import HighlightedValue from './highlightedvalue';
+import { nestValue } from '../utils/objectutils';
+
+/**
+ * Represents highlighted fields with the highlighting applied
+ */
+export default class AppliedHighlightedFields {
+  /**
+   * Constructs a highlighted field map which consists of mappings from fields to highlighted
+   * value strings.
+   *
+   * @param {Object<string, string|object>} data Keyed by fieldName. It may consist of nested fields
+   *
+   * Example object:
+   *
+   * {
+   *   name: '<strong>Yext</strong>',
+   *   description: {
+   *     featured: '<strong>Yext</strong> is the offical answers company'
+   *   }
+   * }
+   */
+  constructor (data = {}) {
+    Object.assign(this, data);
```

Comment:
Do we not know the field names beforehand? It feels like it's hard to reason what this object looks like without actually having one to log.

Reply:
That's correct, we don't know the field names beforehand since live API choses which fields to highlight based on the search query